recipes/libev/all/conanfile.py

```diff
@@ -30,22 +31,25 @@ def configure(self):
             # libtool:   error: can't build i686-pc-mingw32 shared library unless -no-undefined is specified
             raise ConanInvalidConfiguration("libev can't be built as shared on Windows")
 
+    def build_requirements(self):
+        if tools.os_info.is_windows and "CONAN_BASH_PATH" not in os.environ:
+            self.build_requires("msys2/20190524")
+
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_folder = "libev-{0}".format(self.version)
         os.rename(extracted_folder, self._source_subfolder)
 
+    @lru_cache(1)
```

Comment:
What a trick! Didn't know about this

Reply:
On python 3.8, it is possible to use `@lru_cache` (without arguments)
but on python 3.7 in mingw this caused a problem.
So I added 1 as argument. Maybe None would also be fine.

Reply:
Well, it is an optimization but I don't like the idea of making the recipe incompatible with lower python versions. In fact, I don't think the creation of the autotools object takes so much time, does it?

Reply:
The creation isn't, but the configure step is!
Ok windows, the configure takes even longer due to the msys2 layer.

Reply:
I didn't know about too! I'm curious how fast is it compared to the early version?

Reply:
It depends on how much configuration is done in the configuration?
On Windows, the difference can be up to multiple minutes.
This can increase if the configuration touches some files which causes the compiler to be invoked because of changed dependencies...
Honestly, it is just the same as adding a `_autotools` member variable to a ConanFile class/object.

Reply:
thanks for sharing this!

Reply:
@madebr but if I understood that correctly, this only caches the python object, not the touches to the files, right? Don't you have to run the configure again? Does it preserve the environment vars?

Reply:
It caches only the Autotools object indeed, one time for each ’conan create’ invocation.
This object will keep all its attributes.
As this decoration function is part of ’functools’, it has a functional programming idiology (which means a similar global state and the autput depends only on the inputs).

I think that is ok in this situation. A configuration should always return a similar object.

If a recipe must run the ’_configure_autotools’ multiple times in different folders, then another approach should be taken. But this is the exception.

Reply:
far enough

Reply:
After some internal discussion, we would like to not use this decorator in recipes as it will restrict the python version and its purpose might be confusing for people not familiar with it

Reply:
Ok.

Closing this as the `msys2` build requirement was merged already.